### DISCUSSION TITLE
define subnet size with get_available_prefixes() in netbox script

### DISCUSSION THREAD

**Discussion by JohnnyWalker2020** [netbox-community/netbox#11605] [Q&A]
Im pretty new to this still, so go easy on me.

Im trying to create a /30 out of a /24 prefix using the Netbox Scripts.  I can do this using the REST_API, but im trying to do this in the Netbox Scripts.

When I use Prefix.get_available_prefixes() it gives me a list of all available prefixes of all prefix lengths.

How do I define the next available /30 subnet???

**Comment by kyerlasswell** [comment]
Did you ever figure this out? Currently trying to do this too.

edit:
For now I just ended up doing something like this, but if there's a better way, that would be good to know about. 

To get the next available, use get_first_available_prefix()

        
        PFX_CIDR = "/30"  # Define the netmask somewhere as a string.
        first_pfx = Prefix.objects.get(id=50).get_first_available_prefix()  # get first available prefix from some parent

        new_child_pfx = Prefix(
            prefix=f"{first_pfx.network.format()}{PFX_CIDR}",  # mash the prefix string and the cidr string together
        )

`first_pfx.network` (or `first_pfx.ip`, I don't think it matters here) returns an IPAddress object without the subnet mask.

`.format()` converts that IPAddress object into a simple string that you can then combine with the subnet mask string. This results in valid input for creating a prefix object.
